### ISSUE TITLE
Keyed Custom Arguments not working

### ISSUE THREAD

**Issue by SHsuperCM** [JorelAli/1.13-Command-API#127]
**CommandAPI version**
3.4

**Minecraft version**
1.16.1

**Describe the bug**
When executed returns this message:
"An unexpected error occurred trying to execute that command"
This might be me not using the api correctly but I'm having trouble with keyed custom arguments, by debugging I found that the CommandAPI doesnt even get to the custom argument parser to check validity so I think this might be some issue within CommandAPI.

**My code**
This code reproduces the behavior:
```java
    private static CustomArgument<TwistedItem> twistedItemArgument() {
        CustomArgument<TwistedItem> twistedItemCustomArgument = new CustomArgument<>((s) -> {
            TwistedItem twistedItem = TwistedItems.INSTANCE.items.get(s);
            if(twistedItem == null)
                throw new CustomArgument.CustomArgumentException("No such TwistedItem!");
            return twistedItem;
        }, true);
        twistedItemCustomArgument.overrideSuggestions(TwistedItems.INSTANCE.items.getIDs().toArray(new String[0]));
        return twistedItemCustomArgument;
    }
```
```java
            LinkedHashMap<String, Argument> arguments;
            arguments = new LinkedHashMap<>();
            arguments.put("command", new LiteralArgument("give"));
            arguments.put("target", new EntitySelectorArgument(EntitySelectorArgument.EntitySelector.MANY_PLAYERS));
            arguments.put("item", twistedItemArgument());
            new CommandAPICommand("twisteditems")
                    .withPermission(CommandPermission.OP)
                    .withArguments(arguments)
                    .executes((sender, args) -> {
                        for (Player player : ((Collection<Player>) args[0]))
                            player.getInventory().addItem(((TwistedItem) args[1]).create());
                    })
                    .register();
```

**Comment by JorelAli** [comment]
Thanks for bringing this to my attention - I'll make sure to investigate this before the 4.0 release later this week

**Comment by JorelAli** [comment]
Hi, I've had the time to look into this issue.
Can confirm it's a bug on my behalf - this is because CustomArgument should be using a MinecraftKey as the function parameter instead of a String. In other words, it wants:
```java
new CustomArgument<>((MinecraftKey key) -> ...);
```

The CommandAPI doesn't even support that! How I missed that, I'll never know. 
Again, thanks for pointing this out!

**Comment by JorelAli** [comment]
Fixed in commit 06b7c1a.
